src/components/App.js

```diff
@@ -18,6 +18,14 @@ const App = () => {
     setInput('')
   }
 
+  const updateToDo = (id, texto) => {
+    const toDoToChange = todos.find((todo) => todo.id === id)
+    const toDoIndex = todos.indexOf(toDoToChange)
+    const newToDos = [...todos]
+    newToDos.splice(toDoIndex, 1, { id: id, text: texto })
```

Comment:
Esto se puede simplificar muchísimo usando un map. La idea es mapear los todos actuales, si el id no coincide con el del parámetro, devolvemos el todo sin modificarlo, si el id coincide, devolvemos el todo modificado

```suggestion
    const updatedTodos = todos.map(todo => todo.id === id ? {...todo, text: texto} : todo )
```